### PR TITLE
Fix various issues

### PR DESCRIPTION
**Pull request by gabor-boros**
**Reason for the change**

We had various smaller issues across project. This PR will fix those issues and gives a bit better config handling.

**Description**

* Pull request body generation moved to the common ``GitMixin`` class
* Pillar will always create lock file in the working directory
* Call expandvar and expanduser of configuration files
* Hammurabi only works in the current working directory
* Read settings (pyproject.toml) path from ``HAMMURABI_SETTINGS_PATH`` environment variable
* Remove faulty author of git committing
* Only attempt to create a PR if there is no PR from Hammurabi 
* Fix double committing issue
* Fix committing of laws when nothing changed
* Fixed several CLI arguments related issues
* Removed target directory setting from config and CLI

**Code examples**

N/A

**Checklist**

- [x] Unit tests created/updated
- [x] Documentation extended/updated
~- [ ] Stubs created/updated~

**References**

N/A

### CODE REVIEW

File: hammurabi/config.py [hunk 1]
```diff
@@ -182,19 +184,62 @@ def __merge_settings(self, loaded_settings: Dict[str, Any]) -> Settings:
...
+    def __get_fallback_repository(self) -> str:
+        """
+        Figure out the fallback owner/repository based on the remote url of the git repo.
+        :return: Returns the owner/repository pair
+        :rtype: str
+        """
+
+        repo_url: str = self.repo.remote().url
+
+        if re.match(r"^http(s)?://", repo_url):
```
Comment (uvegla) [inline hammurabi/config.py:196]: I think a simple `repo_url.startswith('http://') or repo_url.startswith('https://')` is slightly more readable here.

File: hammurabi/mixins.py [hunk 2]
```diff
@@ -140,6 +137,10 @@ def generate_pull_request_body(pillar) -> str:
         #  NOTE: The parameter type can not be hinted, because of circular import,
         #  we must fix this in the future releases.
 
+        def get_chained_rules(target, items):
```
Comment (uvegla) [inline hammurabi/mixins.py]: I would use `chain_rules`. Chained may also be correct. Tho I would only change it if we use the same terminology everywhere.

**Review by uvegla — DISMISSED** [review]

File: hammurabi/config.py (hunk 1 above)
Comment (gabor-boros) [reply hammurabi/config.py:196]: Hm, maybe yes. The only concern I have against the `startswith` is that it's longer, which could make the code less readable on the long run. WDYT?

Comment (uvegla) [reply hammurabi/config.py:196]: I am fine with the regex as well. 😃👍

**Review by uvegla — APPROVED** [review]

**Pull request merged**